anyway/infographics_utils.py

```diff
@@ -158,6 +166,7 @@ def serialize(self):
 
 class WidgetCollection:
     widgets: List[Type[Widget]] = []
+    widgets_dict: Dict[str, Widget] = {}
```

Comment:
This should be `Dict[str, Type[Widget]]`.

We don't need `widgets` anymore, we can simply iterate `widgets_dict.values()` (though we can call it `widgets`.

BTW, there's no need for `WidgetCollection` if it's never instantiated. We can have `widgets_dict` as a global variable (which is what `WidgetCollection` is, effectively), no constructor, no `get()` method, just a global dictionary and a global `register_widget` function.

And if we do have the class, no need for the `get()` method - just access the variable. `get()` is only needed if it's an expensive access that we want client code to be aware of. Languages with `@property` don't need getters.

Reply:
I agree `widgets` is not needed.
Regarding `WidgetCollection`, I prefer to have the class to enclose the implementation of the gathering of the widgets. (I will prefix the dict with `__`)

Reply:
IMO there's no need for that kind of encapsulation, and it's not standard to use it in Python (the implementation is exposed just the same with the `get()` method). It makes it look more complex than it is.

Reply:
Using a method to get a widget class from its name will allow us to change the implementation in the future with less code changes. If it is OK with you, I prefer to leave the class, if not, I will remove it.

Reply:
This is true in Java. Python (and C#, and many other languages) has `@property` methods so the difference is invisible to the client code; the function call syntax hints the reader that this call might have a side effect (or that it is computationally heavy) - which is not the case here.

Also, everything here is intended for in-module use, so this kind of change will be easy; and if it is not, the fact that we return a mutable collection exposes much more significant implementation detail and is prone to misuse.

Of course I'm not here to dictate anything, even if I could :) I'm just saying what I think.